pkg/docker/docker.go

```diff
@@ -310,14 +318,20 @@ func NewEngineAPIClient(config *api.DockerConfig) (*dockerapi.Client, error) {
 			},
 		}
 	}
-	return dockerapi.NewClient(config.Endpoint, os.Getenv("DOCKER_API_VERSION"), httpClient, nil)
+
+	return dockerapi.NewClientWithOpts(
+		dockerapi.WithHost(config.Endpoint),
+		dockerapi.WithHTTPClient(httpClient),
```

Comment:
Are these even need today? AFAIK just using `dockerapi.FromEnv` should ensure everything is set up just all right.